rpc/client.go

```diff
@@ -119,26 +127,35 @@ func (client *Client) runWorkers(n int) {
 }
 
 // handleRequest handles writing the request for each Darknode to the queue and reading each of the responses.
-func (client *Client) handleRequest(request interface{}, method string, addresses []string) []interface{} {
+func (client *Client) handleRequest(request interface{}, method string, addresses []addr.Addr) []interface{} {
 	results := make([]interface{}, len(addresses))
 
 	// Loop through the provided addresses.
 	co.ParForAll(addresses, func(i int) {
```

Comment:
The overhead of using `ParForAll` here is most likely going to be greater than processing this sequentially. At most, I would use a `ForAll` (which will only create one goroutine per CPU, instead of one goroutine per address). There's no IO happening in this processing (we are just preparing calls and dumping them on a queue for other workers to handle, which is the right approach) and no unbounded blocking (all the potentially blocking operations are guaranteed to finish in reasonable time or result in reasonable back-pressure) so using `ParForAll` is overkill.